tracer/src/Datadog.Trace/Debugger/Symbols/SymbolExtractor.cs

```diff
@@ -0,0 +1,718 @@
+
+                var methodDef = MetadataReader.GetMethodDefinition(methodDefHandle);
+                var methodScope = CreateMethodScope(type, methodDef);
+                if (methodScope.Scopes != null &&
+                    methodScope is { StartLine: UnknownMethodStartLine, EndLine: UnknownMethodEndLine, SourceFile: null })
+                {
+                    var sourcePdbInfo = GetMethodSourcePdbInfo(methodScope);
+                    methodScope.StartLine = sourcePdbInfo.StartLine;
+                    methodScope.EndLine = sourcePdbInfo.EndLine;
+                    methodScope.SourceFile = sourcePdbInfo.Path;
+                    if (sourcePdbInfo.EndColumn > 0)
+                    {
+                        var ls = new LanguageSpecifics
+                        {
+                            Annotations = methodScope.LanguageSpecifics?.Annotations,
+                            AccessModifiers = methodScope.LanguageSpecifics?.AccessModifiers,
+                            ReturnType = methodScope.LanguageSpecifics?.ReturnType,
+                            StartColumn = sourcePdbInfo.StartColumn,
+                            EndColumn = sourcePdbInfo.EndColumn,
+                        };
+
+                        methodScope.LanguageSpecifics = ls;
+                    }
+                }
+
+                classMethods[index] = methodScope;
+                index++;
+            }
+        }
+
+        private SourcePdbInfo GetMethodSourcePdbInfo(Model.Scope methodScope)
+        {
+            var startLine = int.MaxValue;
+            var endLine = 0;
+            int? startColumn = null;
+            int? endColumn = null;
+            string? sourceFile = null;
+            for (int i = 0; i < methodScope.Scopes.Count; i++)
+            {
+                var scope = methodScope.Scopes[i];
+                if (startLine > scope.StartLine && scope.StartLine > 0)
+                {
+                    startLine = scope.StartLine;
+                }
+
+                if (endLine < scope.EndLine)
+                {
+                    endLine = scope.EndLine;
+                }
+
+                sourceFile ??= scope.SourceFile;
+
+                if (!scope.LanguageSpecifics.HasValue)
+                {
+                    continue;
+                }
+
+                if (startColumn == null || startColumn > scope.LanguageSpecifics.Value.StartColumn)
+                {
+                    startColumn = scope.LanguageSpecifics.Value.StartColumn;
+                }
+
+                if (endColumn == null || endColumn < scope.LanguageSpecifics.Value.EndColumn)
+                {
+                    endColumn = scope.LanguageSpecifics.Value.EndColumn;
+                }
+            }
+
+            if (startLine == int.MaxValue)
+            {
+                startLine = UnknownMethodStartLine;
+            }
+
+            if (endLine == 0)
+            {
+                endLine = UnknownMethodEndLine;
+            }
```

Comment:
Can we simply set the `startLine` and `endLine` to the lines of the source file itself?
And what about the value of `StartColumn` and `EndColumn` - are we ok with them potentially holding the value `null`?

Reply:
1. We don't know them but for our needs we try to set them to first and end legally sequence points and if we fail, to a known sign so the backend will know what to do.
2. Good point, we handle that for now but potentially we have an issue in some cases.

Reply:
What is the known sign agreed upon? 🤔 
I won't resolve this conversation but don't consider it as a blocker. I would say let's not make it a blocker but revisit it once your PR is merged.

Reply:
It is written in the internal RFC